parse_helpers_test.go

```diff
@@ -0,0 +1,151 @@
+package memefish_test
+
+import (
+	"fmt"
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/cloudspannerecosystem/memefish"
+)
+
+func ExampleParseStatement() {
+	stmt, err := memefish.ParseStatement("path/to/file.sql", "SELECT * FROM foo")
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(stmt.SQL())
+
+	// Output:
+	// SELECT * FROM foo
+}
+
+func ExampleParseStatements() {
+	stmts, err := memefish.ParseStatements("path/to/file.sql", "SELECT 1; INSERT foo (x, y) VALUES (1, 2)")
+	if err != nil {
+		panic(err)
+	}
+
+	for _, stmt := range stmts {
+		fmt.Printf("%s;\n", stmt.SQL())
+	}
+
+	// Output:
+	// SELECT 1;
+	// INSERT INTO foo (x, y) VALUES (1, 2);
+}
+
+func ExampleParseQuery() {
+	stmt, err := memefish.ParseQuery("path/to/file.sql", "SELECT * FROM foo")
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(stmt.SQL())
+
+	// Output:
+	// SELECT * FROM foo
+}
+
+func ExampleParseExpr() {
+	stmt, err := memefish.ParseExpr("", "a * b")
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(stmt.SQL())
+
+	// Output:
+	// a * b
+}
+
+func ExampleParseType() {
+	stmt, err := memefish.ParseType("", "ARRAY<STRUCT<n INT64, s STRING>>")
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(stmt.SQL())
+
+	// Output:
+	// ARRAY<STRUCT<n INT64, s STRING>>
+}
+func ExampleParseDDL() {
+	sql := heredoc.Doc(`
+					CREATE TABLE foo (
```

Comment:
It seems better to fix indentation of heredoc strings

Reply:
I think so too.

Reply:
fixed https://github.com/cloudspannerecosystem/memefish/pull/148/commits/d1157ea6a83eb4930dd7fd65b37f9e7e365fcfb4

Reply:
I don't know we can agree with indent level with different tabstop settings.
(I only use Jetbrains Goland.)